### DISCUSSION TITLE
Expose 'close' functionality for Listbox(and other) components

### DISCUSSION THREAD

**Discussion by Joostdeboer** [tailwindlabs/headlessui#1775] [Ideas]
At the moment of writing this, the only headlessui component that exposes a 'close' functionality to the user, such that you can manually close the component, is the Disclosure. However, while all the components have an 'open' boolean, they do not have access to a function to close the component, if applicable. For example, the Listbox would be one such component where having access to the 'close' function would be beneficial for developing custom elements.

In our use-case we would like to close the Listbox component on scrolling through the webpage, and while we can find out whether a user has scrolled through listening to the document's scroll in a useEffect hook, we cannot close the Listbox itself through this function. We think it would benefit developers to allow for this behavior.

**Comment by itsjms** [comment]
@Joostdeboer We did this https://github.com/tailwindlabs/headlessui/discussions/1779#discussioncomment-3503138 for the Popover component. Hope it helps.
